internal/bminventory/inventory.go

```diff
@@ -3114,8 +3114,12 @@ func (b *bareMetalInventory) V2DeregisterHostInternal(ctx context.Context, param
 	}
 
 	// TODO: need to check that host can be deleted from the cluster
-	b.eventsHandler.AddEvent(ctx, params.InfraEnvID, &params.HostID, models.EventSeverityInfo,
-		fmt.Sprintf("Host %s: deregistered from cluster", params.HostID.String()), time.Now())
+	infraEnv, err := common.GetInfraEnvFromDB(b.db, params.InfraEnvID)
+	var clusterID strfmt.UUID
+	if err != nil && infraEnv != nil {
```

Comment:
this condition is not possible: if GetInfraEnvFromDB returns an error, then infraEnv will be nil

Reply:
fixed